vertx-config/src/test/java/io/vertx/config/impl/spi/SystemPropertiesConfigStoreTest.java

```diff
@@ -41,6 +41,14 @@ public void init() {
     System.setProperty("float", "25.3");
     System.setProperty("true", "true");
     System.setProperty("false", "false");
+
+    System.setProperty("hierarchical.key", "value");
```

Comment:
where is the result asserted ?

Reply:
This is asserted by the call to `ConfigChecker.check` on line 95

Reply:
I also added some more assertions to the flat parsing case (lines 79-88)